src/audio/google_rtc_audio_processing.c

```diff
@@ -309,15 +318,18 @@ static int google_rtc_audio_processing_copy(struct comp_dev *dev)
 
 	aec_reference_buff_frag = 0;
 	for (frame = 0; frame < num_aec_reference_frames; frame++) {
-		src = audio_stream_read_frag_s16(&cd->aec_reference->stream,
-										 aec_reference_buff_frag);
-		cd->aec_reference_buffer[cd->aec_reference_buffer_index] = *src;
-		++cd->aec_reference_buffer_index;
+		for (channel = 0; channel < cd->num_aec_reference_channels; ++channel) {
+			src = audio_stream_read_frag_s16(&cd->aec_reference->stream,
```

Comment:
The frag API is being deprecated by @singalsu as it adds MCPS. Do you want to merge this now and fix the frag API usage later ?

Reply:
I think we should do that. 
I want to get rid of the frag API but I'll have to do that carefully to avoid buffering issue.
The reason the frag API works here is that sometimes the aec_reference has 4 channels, using the frag API allows easily to only use those 2 first ones).